pulp_2to3_migration/tests/functional/test_file_migration.py

```diff
@@ -77,25 +93,33 @@ def setUpClass(cls):
         cls.tasks_api = TasksApi(core_client)
         cls.migration_plans_api = MigrationPlansApi(migration_client)
 
-    def test_migrate_iso_repositories(self):
-        """Test that a Migration Plan to migrate ISO content executes correctly."""
-        migration_plan = '{"plugins": [{"type": "iso"}]}'
+    def _do_test(self, repos, migration_plan):
         mp = self.migration_plans_api.create({'plan': migration_plan})
         mp_run_response = self.migration_plans_api.run(mp.pulp_href, data={})
         task = monitor_task(self.tasks_api, mp_run_response.task)
         self.assertEqual(task.state, "completed")
-        pulp2repositories = Pulp2Repository.objects.all()
-        for pulp2repo in pulp2repositories:
-            pulp3repos = self.file_repo_api.list(name=pulp2repo.repo_id)
-            repo_href = pulp3repos.results[0].pulp_href
+        for repo_id in repos:
+            pulp3repos = self.file_repo_api.list(name=repo_id)
+            # Assert that there is a result
             self.failIf(not pulp3repos.results,
                         "Missing a Pulp 3 repository for Pulp 2 "
-                        "repository id '{}'".format(pulp2repo.repo_id))
+                        "repository id '{}'".format(repo_id))
+            repo_href = pulp3repos.results[0].pulp_href
             # Assert that the name in pulp 3 matches the repo_id in pulp 2
-            self.assertEqual(pulp2repo.repo_id, pulp3repos.results[0].name)
+            self.assertEqual(repo_id, pulp3repos.results[0].name)
             # Assert that there is a Repository Version with the same number of content units as
             # associated with the repository in Pulp 2.
             repo_version_href = self.file_repo_versions_api.list(repo_href).results[0].pulp_href
-            pulp2_repo_content = RepositoryContentUnit.objects.filter(repo_id=pulp2repo.repo_id)
             repo_version_content = self.file_content_api.list(repository_version=repo_version_href)
-            self.assertEqual(pulp2_repo_content.count(), repo_version_content.count)
+            self.assertEqual(PULP_2_ISO_FIXTURE_DATA[repo_id], repo_version_content.count)
+        self.assertEqual(len(repos), self.file_repo_api.list().count)
+
+    def test_1_migrate_specific_iso_repositories(self):
+        """Test that a Migration Plan with repos specified migrates only those repos."""
+        repos = list(PULP_2_ISO_FIXTURE_DATA.keys())[:2]
+        self._do_test(repos, SPECIFIC_REPOS_MIGRATION_PLAN)
```

Comment:
if we want to compare the results with the specific plan, then i think PULP_2_ISO_FIXTURE_DATA should be ordereddict otherwise the order of the ( currently 4 repos?) is not guaranteed.

Reply:
I don't think that's the case.  It just selects the first two repo_id, and which ones they are doesn't actually matter which ones specifically because it looks up the count later using the repo_id.  So even picking them at random should be fine.

But it should always be the same ones because  as of Python 3.6 ```dict``` is ordered, and as of 3.7 it was made part of the spec (whereas before it was an implementation detail).